File: backend/app/schemas/travel.py
from datetime import datetime
from typing import Any, Optional
from pydantic import BaseModel, Field


class TravelPlanCreate(BaseModel):
    """创建旅行计划"""
    title: str = Field(..., max_length=200, description="计划标题")
    destination: str = Field(..., max_length=100, description="目的地")
    days: int = Field(..., ge=1, description="旅行天数")
    budget: Optional[float] = Field(None, ge=0, description="预算")
    plan_json: Optional[dict] = Field(None, description="结构化行程JSON")
    markdown: Optional[str] = Field(None, description="AI生成的可展示Markdown文本")


class TravelPlanUpdate(BaseModel):
    """更新旅行计划"""
    title: Optional[str] = Field(None, max_length=200)
    destination: Optional[str] = Field(None, max_length=100)
    days: Optional[int] = Field(None, ge=1)
    budget: Optional[float] = Field(None, ge=0)
    plan_json: Optional[dict] = None
    markdown: Optional[str] = None


class TravelPlanResponse(BaseModel):
    """旅行计划响应"""
    id: int
    user_id: int
    title: str
    destination: str
    days: int
    budget: Optional[float] = None
    plan_json: Optional[Any] = None
    markdown: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/travel.ts
/**
 * 旅游资源相关类型定义（对应后端 app/schemas/city.py, scenic.py, hotel.py, restaurant.py, travel.py）
 */

// ==================== City ====================

export interface CityResponse {
  id: number
  name: string
  province: string
  country: string
  description: string | null
  cover_image: string | null
  latitude: number | null
  longitude: number | null
  level: string // 热门 / 普通 / 小众
}

// ==================== ScenicSpot ====================

export interface ScenicSpotResponse {
  id: number
  city_id: number
  name: string
  description: string | null
  address: string | null
  category: string | null
  score: number | null
  price: number | null
  open_time: string | null
  latitude: number | null
  longitude: number | null
  image_url: string | null
  tags_json: Record<string, unknown> | null
}

// ==================== Hotel ====================

export interface HotelResponse {
  id: number
  city_id: number
  name: string
  description: string | null
  address: string | null
  price: number | null
  score: number | null
  open_time: string | null
  latitude: number | null
  longitude: number | null
  image_url: string | null
}

// ==================== Restaurant ====================

export interface RestaurantResponse {
  id: number
  city_id: number
  name: string
  category: string | null
  description: string | null
  price_level: string | null
  score: number | null
  address: string | null
  latitude: number | null
  longitude: number | null
  image_url: string | null
}

// ==================== TravelPlan ====================

export interface TravelPlanResponse {
  id: number
  user_id: number
  title: string
  destination: string
  days: number
  budget: number | null
  plan_json: Record<string, unknown> | null
  markdown: string | null
  created_at: string
  updated_at: string
}
